### DISCUSSION TITLE
Why do derived atoms cause double rerenders every other update?

### DISCUSSION THREAD

**Discussion by elliotwaite** [pmndrs/jotai#1218] [Q&A] [answered]
Here's a [demo](https://codesandbox.io/s/jotai-derived-atom-render-count-test-dxsnfr?file=/src/App.jsx).

Clicking the "increment count" button causes the derived count component to rerender twice every other update.

Screenshot:
![Frame 1](https://user-images.githubusercontent.com/1767836/172742732-8a41eaae-0af7-4195-9cbd-58ef0e2c3ede.png)

Is this expected behavior?

**Comment by dai-shi** [comment, marked as answer]
This is kind of expected and new behavior in React 18. https://twitter.com/dai_shi/status/1534170089981100032

In React 17, it was fine. But, it's not ideal in React 18.
#1214 will fix it: https://codesandbox.io/s/jotai-derived-atom-render-count-test-forked-1ndf5u?file=/src/App.jsx
#1159 is also a fix for React 18 behavior.

**Reply by elliotwaite** [reply]
Okay, nice. Thanks for the explanation.

**Comment by meddyrainzo** [comment]
Has this particularly been fixed? Just started a new project and getting double renders all the time

**Reply by dai-shi** [reply]
If you still see double renders, it's probably for derived atoms. It is an intended behavior. If this behavior causes serious performance issue, wrap your heavy computation with `useMemo`. (We should probably have this tip in docs, if we don't yet.)
